### PR TITLE
WIP: finishing ext wod

### PR DESCRIPTION
**Pull request by nchronas**
@surligas @squarewavesbelow @zisi @mashua Please see the comments and let me know if it compiles to your systems.

### CODE REVIEW

File: platform/adcs/housekeeping.c [hunk 1]
```diff
@@ -25,13 +25,45 @@ SAT_returnState hk_report_parameters(HK_struct_id sid, tc_tm_pkt *pkt) {
...
 
         uint16_t size = 1;
+        uint32_t sys_epoch = 0;
         /* System specific */
         cnv32_8(HAL_sys_GetTick(), &pkt->data[size]);
         size += 4;
+        get_time_QB50(&sys_epoch);
+        cnv32_8(sys_epoch, &pkt->data[size]);
+        size += 4;
+        pkt->data[size] = adcs_data.rsrc;
```
Comment (nchronas) [inline platform/adcs/housekeeping.c]: @zisi Is this the correct variable that you store the reset source?

File: platform/comms/housekeeping.c [hunk 2]
```diff
@@ -45,6 +45,16 @@ SAT_returnState hk_report_parameters(HK_struct_id sid, tc_tm_pkt *pkt) {
 	i = 1;
         cnv32_8(HAL_sys_GetTick(), &pkt->data[i]);
         i += sizeof(uint32_t);
+        /* FIXME: Get the actual reboot code */
+        b = comms_stats.rst_src;
```
Comment (nchronas) [inline platform/comms/housekeeping.c:49]: @surligas  Is this the correct variable that you store the reset source?

File: platform/eps/housekeeping.c [hunk 3]
```diff
@@ -133,6 +133,13 @@ SAT_returnState hk_report_parameters(HK_struct_id sid, tc_tm_pkt *pkt) {
         cnv32_8( HAL_sys_GetTick(), &pkt->data[1]);
         size += 4;
 
+        //Please add the correct variable
+        pkt->data[size] = rsrc;
```
Comment (nchronas) [inline platform/eps/housekeeping.c:137]: @squarewavesbelow  Please let me know the correct variable that you store the reset source.

File: platform/obc/housekeeping.c [hunk 4]
```diff
@@ -184,66 +190,25 @@ SAT_returnState hk_parameters_report(TC_TM_app_id app_id, HK_struct_id sid, uint
...
-        //cnv8_F(&data[37], &sat_ext_status.adcs_vsun[2]);
-        //cnv8_F(&data[41], &sat_ext_status.adcs_vsun[3]);
-        //cnv8_F(&data[45], &sat_ext_status.adcs_vsun[4]);
-        //cnv8_F(&data[49], &sat_ext_status.adcs_long_sun);
-        //cnv8_F(&data[53], &sat_ext_status.adcs_lat_sun);
-        //cnv8_32(&data[57], &sat_ext_status.adcs_m_RPM);
-
+        
+        if(!C_ASSERT(len != ADCS_EXT_WOD_SIZE) == true) { return SATR_ERROR; }
+        //Apo take from here the sciende header
```
Comment (nchronas) [inline platform/obc/housekeeping.c]: @mashua Please let me know the struct you use to store the science data

File: services/housekeeping_service.h [hunk 5]
```diff
@@ -3,6 +3,16 @@
...
+//finalize sizes
+
+#define COMMS_EXT_WOD_SIZE   1
+#define COMMS_EXT_WOD_OFFSET 1
+#define ADCS_EXT_WOD_SIZE    1
+#define ADCS_EXT_WOD_OFFSET  1
+#define EPS_EXT_WOD_SIZE     1
+#define EPS_EXT_WOD_OFFSET   1
+
+#define SUB_SYS_EXT_WOD_SIZE 1
```
Comment (nchronas) [inline services/housekeeping_service.h]: @ppapadeas We need to finalize the sizes and add it here.

File: platform/obc/housekeeping.c (hunk 4 above)
Comment (nchronas) [inline platform/obc/housekeeping.c]: done

File: platform/comms/housekeeping.c (hunk 2 above)
Comment (nchronas) [inline platform/comms/housekeeping.c:49]: ok

Comment (surligas) [inline platform/comms/housekeeping.c:49]: Yeap!

Comment (surligas) [inline platform/comms/housekeeping.c:49]: Fixed the boot event byte at 43c38f72a4eeaa6d8442a0ded86afcda6b5cff9a. Now we miss only the last assertion code

**Pull request closed without merging**